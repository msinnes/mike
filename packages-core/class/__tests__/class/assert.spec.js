const BaseClass = require('../../src/class/base');

const assertClass = require('../../src/class/assert');

describe('assertClass', () => {
  it('should be a function', () => {
    expect(assertClass).toBeDefined();
    expect(assertClass).toBeInstanceOf(Function);
  });

  it('should not throw an error if the input is a class', () => {
    expect(() => {
      assertClass(BaseClass);
    }).not.toThrow();
  });

  it('should throw an error if input is not a class', () => {
    expect(() => {
      assertClass(undefined, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertClass(null, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertClass('', 'message');
    }).toThrowErrorMatchingSnapshot();
  });
});
