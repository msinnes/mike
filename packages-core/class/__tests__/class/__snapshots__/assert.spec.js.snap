// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertClass should throw an error if input is not a class 1`] = `"ClassError: message"`;

exports[`assertClass should throw an error if input is not a class 2`] = `"ClassError: message"`;

exports[`assertClass should throw an error if input is not a class 3`] = `"ClassError: message"`;
